shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey1 = "test1." + newId();
+                let newValue1 = newId();
+
+                // First cycle: disable -> cache -> enable -> flush
+                manager.set(newKey1, newValue1);
+                Assert.equal(newValue1, manager.get(newKey1), "Should return cached value");
+
+                configValues.cookieCfg.enabled = true;
+                Assert.equal(newValue1 + "; path=/", this._testCookies[newKey1], "First cookie should be flushed");
+
+                // Second cycle: disable -> cache -> enable -> flush
+                configValues.cookieCfg.enabled = false;
+                let newKey2 = "test2." + newId();
+                let newValue2 = newId();
+
+                manager.set(newKey2, newValue2);
+                Assert.equal(newValue2, manager.get(newKey2), "Should return second cached value");
+
+                configValues.cookieCfg.enabled = true;
+                Assert.equal(newValue2 + "; path=/", this._testCookies[newKey2], "Second cookie should be flushed");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Deletion operation cached and applied via dynamic config change",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: true
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+                
+                // Set a cookie while enabled
+                manager.set(newKey, newValue);
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should be in storage");
+                
+                // Disable cookies via config and delete the cookie
+                configValues.cookieCfg.enabled = false;
+                let delResult = manager.del(newKey);
+                Assert.equal(true, delResult, "Delete should return true when cached");
+                
+                // Cookie still exists in storage but deletion is cached
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should still be in storage while disabled");
+                
+                // Enable cookies via dynamic config change - cached deletion should be applied
+                configValues.cookieCfg.enabled = true;
+                
+                // Check that the deletion was applied
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect